Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -61,6 +62,8 @@ public IdentityDeletionProcess StartDeletionProcessAsSupport()
     {
         EnsureNoActiveProcessExists();
 
+        TierBeforeDeletionId = TierId;
```

Comment:
Mhh. The TierBeforeDeletion should only be saved when the deletion process transitions to Approved - which is not the case when the deletion process is started as support